lib/stripe/util.rb

```diff
@@ -104,19 +104,29 @@ def self.custom_method(resource, target, name, http_verb, http_path)
     # ==== Attributes
     #
     # * +data+ - Hash of fields and values to be converted into a StripeObject.
+    # * +params+ - Params for +StripeObject+ like filters used in search that
+    #   will be reused on subsequent API calls.
     # * +opts+ - Options for +StripeObject+ like an API key that will be reused
     #   on subsequent API calls.
-    def self.convert_to_stripe_object(data, opts = {})
+    def self.convert_to_stripe_object(data, params = {}, opts = {})
       opts = normalize_opts(opts)
 
       case data
       when Array
-        data.map { |i| convert_to_stripe_object(i, opts) }
+        data.map { |i| convert_to_stripe_object(i, {}, opts) }
       when Hash
         # Try converting to a known object class.  If none available, fall back
         # to generic StripeObject
-        object_classes.fetch(data[:object], StripeObject)
-                      .construct_from(data, opts)
+        obj = object_classes.fetch(data[:object], StripeObject)
+                            .construct_from(data, opts)
+
+        # set filters so that we can fetch the same limit, expansions, and
+        # predicates when accessing the next and previous pages
+        if obj && (obj.is_a?(SearchResultObject) || obj.is_a?(ListObject))
+          obj.filters = params
```

Comment:
Previously we dup'd the filters. We should probably continue doing this here right?